src/engine/ui/widgets/Combobox.cxx

```diff
@@ -60,25 +67,47 @@ void ComboBox::setPosition(int x, int y)
   m_uiElementRect.y = y;
 
   // reposition the dropdown menu frame
-  m_dropDownRect.x = x;
-  m_dropDownRect.y = m_uiElementRect.y + m_uiElementRect.h;
+  m_dropdownRect.x = x;
+  m_dropdownRect.y = m_uiElementRect.y + m_uiElementRect.h;
 
   // adjust the rect that represents the whole element
   m_wholeElementRect.x = x;
   m_wholeElementRect.y = y;
 
   // reposition the text field
-  m_textField->setPosition(m_dropDownRect.x, m_dropDownRect.y);
+  m_dropdownRect.x = x;
+  m_dropdownRect.y = y;
+  m_highlightingRect.x = x + 4;
+
+  int currentElement = 0;
+  for (const auto &text : m_items)
+  {
+    //SDL_Rect textRect;
+    switch (textAlignment)
+    {
+    case TextFieldAlignment::RIGHT:
+      x = m_dropdownRect.x + m_dropdownRect.w - text->getUiElementRect().w;
+      break;
+    case TextFieldAlignment::CENTERED:
+      x = m_dropdownRect.x + (m_dropdownRect.w / 2 - text->getUiElementRect().w / 2);
+      break;
+    default: //AKA TextFieldAlignment::LEFT
```

Comment:
this is ambigous, please check for TextFieldAlignment::LEFT and default to make it explicit
```suggestion
    TextFieldAlignment::LEFT:
    default:
```